clara/interpreter.py

```diff
@@ -208,17 +208,23 @@ def execute_ListHead(self, l, mem):
         t = l.args[0].value
         l = self.execute(l.args[1], mem)
 
-        if isinstance(l, list) and len(l) > 0:
-            return self.convert(l[0], t)
+        if isinstance(l, list):
+            if len(l) > 0:
+                return self.convert(l[0], t)
+            elif len(l) == 0:
+                return UndefValue()
 
         raise RuntimeErr("ListHead on '%s'" % (l,))
 
     def execute_ListTail(self, l, mem):
 
         l = self.execute(l.args[0], mem)
 
-        if isinstance(l, list) and len(l) > 0:
-            return list(l[1:])
+        if isinstance(l, list):
+            if len(l) > 0:
+                return list(l[1:])
+            elif len(l) == 0:
```

Comment:
This case earlier caused `RuntimeErr`, now it fails silently. Why should this happen?

Reply:
As I can not remember my codes exactly, I'm not sure, but I think it was a a patch to handle codes like:
```c
while (scanf("%d", &n) > 0)
    m += n;
```
in
https://github.com/iradicek/clara/blob/29e939a1b0dc14a39f08794b6a8bbf62e64e0878/clara/c_interpreter.py#L236